src/azure-cli/azure/cli/command_modules/servicefabric/_params.py

```diff
@@ -278,8 +279,18 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('thumbprint', nargs='+', help='A single or Space-separated list of client certificate thumbprint(s) to be remove.')
         c.argument('common_name', nargs='+', help='A single or Space-separated list of client certificate common name(s) to be remove.')
 
+    with self.argument_context('sf managed-cluster network-security-rule add', validator=validate_add_network_security_rule) as c:
+        c.argument('name', help='Network security rule name')
+        c.argument('access', help='possible values are <allow> or <deny>')
+        c.argument('direction', help='possible values are <inbound> or <outbound>')
+        c.argument('description', help='network security rule description')
+        c.argument('priority', help='integer that shows priority for rule')
+        c.argument('protocol', help='enter one of the following: tcp, htpps, http, udp, icmp, ah, esp, any')
+        c.argument('source_port_ranges', options_list=['--source-port-ranges'], nargs='+', help='A single or space separated list of source port ranges')
+        c.argument('dest_port_ranges', options_list=['--dest-port-ranges'], nargs='+', help='A single or space separated list of destination port ranges')
+        c.argument('source_addr_prefixes', options_list=['--source-addr-prefixes'], nargs='+', help='A single or space separated list of source address prefixes')
+        c.argument('dest_addr_prefixes', options_list=['--dest-addr-prefixes'], nargs='+', help='A single or space separated list of destination address prefixes')
```

Comment:
The default option name does not need to be explicitly specified

```suggestion
        c.argument('source_port_ranges', nargs='+', help='A single or space separated list of source port ranges')
        c.argument('dest_port_ranges', nargs='+', help='A single or space separated list of destination port ranges')
        c.argument('source_addr_prefixes', nargs='+', help='A single or space separated list of source address prefixes')
        c.argument('dest_addr_prefixes', nargs='+', help='A single or space separated list of destination address prefixes')
```